sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/transforms/ParDoTest.java

```diff
@@ -1523,20 +1528,66 @@ public void testMutatingInputCoderDoFnError() throws Exception {
   }
 
   @Test
-  public void testIncludesDoFnDisplayData() {
-    Bound<String, String> parDo =
-        ParDo.of(
-            new DoFn<String, String>() {
-              @Override
-              public void processElement(ProcessContext c) {}
+  public void testDoFnDisplayData() {
+    DoFn<String, String> fn = new DoFn<String, String>() {
+      @Override
+      public void processElement(ProcessContext c) {
+      }
 
-              @Override
-              public void populateDisplayData(Builder builder) {
-                builder.add("foo", "bar");
-              }
-            });
+      @Override
+      public void populateDisplayData(Builder builder) {
+        builder.add("doFnMetadata", "bar");
+      }
+    };
+
+    Bound<String, String> parDo = ParDo.of(fn);
+
+    DisplayData displayData = DisplayData.from(parDo);
+    assertThat(displayData, hasDisplayItem(allOf(
+        hasKey("fn"),
+        hasType(DisplayData.Type.JAVA_CLASS),
+        hasValue(fn.getClass().getName()))));
+
+    assertThat(displayData, includes(fn));
+  }
+
+  @Test
+  public void testDoFnWithContextDisplayData() {
+    DoFnWithContext<String, String> fn = new DoFnWithContext<String, String>() {
+      @ProcessElement
+      public void proccessElement(ProcessContext c) {}
+
+      @Override
+      public void populateDisplayData(Builder builder) {
+        builder.add("fnMetadata", "foobar");
+      }
+    };
+
+    Bound<String, String> parDo = ParDo.of(fn);
+
+    DisplayData displayData = DisplayData.from(parDo);
+    assertThat(displayData, includes(fn));
```

Comment:
Maybe add some messages here? Every time I see these tests, I expect to see something like "and make sure the fnMetadata" is there, then I need to find this and parse it.

`assertThat("Copies over all data from the function", displayData, includes(fn));`

Might make it easier to parse the assertions.